ush/forecast_predet.sh

```diff
@@ -224,26 +227,70 @@ FV3_predet(){
       ${NLN} "${file}" "${COM_ATMOS_RESTART}/${forecast_end_cycle:0:8}.${forecast_end_cycle:8:2}0000.${file}"
     done
   else
-    mkdir -p "${DATA}/RESTART"
+    if [[ ! -d "${DATA}/RESTART" ]]; then mkdir -p "${DATA}/RESTART"; fi
   fi
 
-  echo "SUB ${FUNCNAME[0]}: pre-determination variables set"
 }
 
 WW3_predet(){
   echo "SUB ${FUNCNAME[0]}: WW3 before run type determination"
+
+  if [[ ! -d "${COM_WAVE_HISTORY}" ]]; then mkdir -p "${COM_WAVE_HISTORY}"; fi
   if [[ ! -d "${COM_WAVE_RESTART}" ]]; then mkdir -p "${COM_WAVE_RESTART}" ; fi
+
   ${NLN} "${COM_WAVE_RESTART}" "restart_wave"
 }
 
 CICE_predet(){
   echo "SUB ${FUNCNAME[0]}: CICE before run type determination"
+
+  if [[ ! -d "${COM_ICE_HISTORY}" ]]; then mkdir -p "${COM_ICE_HISTORY}"; fi
+  if [[ ! -d "${COM_ICE_RESTART}" ]]; then mkdir -p "${COM_ICE_RESTART}"; fi
+  if [[ ! -d "${COM_ICE_INPUT}" ]]; then mkdir -p "${COM_ICE_INPUT}"; fi
+
   if [[ ! -d "${DATA}/CICE_OUTPUT" ]]; then  mkdir -p "${DATA}/CICE_OUTPUT"; fi
   if [[ ! -d "${DATA}/CICE_RESTART" ]]; then mkdir -p "${DATA}/CICE_RESTART"; fi
+
+  # CICE does not have a concept of high frequency output like FV3
+  CICE_OUTPUT_FH=""
+  local fhr
+  for (( fhr = FHMIN; fhr <= FHMAX; fhr = fhr + FHOUT )); do
+    CICE_OUTPUT_FH="${CICE_OUTPUT_FH} ${fhr}"
+  done
+
 }
 
 MOM6_predet(){
   echo "SUB ${FUNCNAME[0]}: MOM6 before run type determination"
+
+  if [[ ! -d "${COM_OCEAN_HISTORY}" ]]; then mkdir -p "${COM_OCEAN_HISTORY}"; fi
+  if [[ ! -d "${COM_OCEAN_RESTART}" ]]; then mkdir -p "${COM_OCEAN_RESTART}"; fi
+  if [[ ! -d "${COM_OCEAN_INPUT}" ]]; then mkdir -p "${COM_OCEAN_INPUT}"; fi
+
   if [[ ! -d "${DATA}/MOM6_OUTPUT" ]]; then mkdir -p "${DATA}/MOM6_OUTPUT"; fi
   if [[ ! -d "${DATA}/MOM6_RESTART" ]]; then mkdir -p "${DATA}/MOM6_RESTART"; fi
+
+  # MOM6 does not have a concept of high frequency output like FV3
+  MOM6_OUTPUT_FH=""
+  local fhr
+  for (( fhr = FHMIN; fhr <= FHMAX; fhr = fhr + FHOUT )); do
+    MOM6_OUTPUT_FH="${MOM6_OUTPUT_FH} ${fhr}"
+  done
```

Comment:
Following from #2264 
```suggestion
  MOM6_OUTPUT_FH=$(seq -s ' ' "${FHMIN}" "${FHOUT}" "${FHMAX}")
```